archive/archivist/trigger_archive_bag.py

```diff
@@ -57,23 +72,36 @@ def publish_messages(topic_arn, messages):
         assert response_status == 200, response
 
 
-def main():
-    args = docopt.docopt(__doc__)
-
-    bags = args['<BAG>']
-    bucket_name = args['--bucket']
-    messages = archive_bag_messages(bags, bucket_name)
-
-    topic_name = args['--topic']
-    # print(f'topic: {topic_name} bucket: {bucket_name}')
+def publish_to_sns(bucket_name, bags, topic_name):
     topic_arn = build_topic_arn(topic_name)
 
     publish_messages(
         topic_arn=topic_arn,
-        messages=messages
+        messages=archive_bag_sns_messages(bags, bucket_name)
     )
 
 
+def call_ingest_api(bucket_name, bags, api):
+    for message in archive_bag_api_messages(bags, bucket_name):
+        response = requests.post(api, json=message)
+        if response.status_code != 200:
+            print(f"Error: [{response.status_code}] {response.json()}")
```

Comment:
Is continuing silently the correct response here? Or would it be better to error explicitly?

Using a requests Session might improve performance slightly – you get a shared connection pool. Similar API:

```python
sess = requests.Session()
for message in archive_bag_api_messages(bags, bucket_name):
    response = sess.post(api, json=message)
```